packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -72,6 +75,58 @@ export function Comments( {
 		);
 	}, [] );
 
+	const { selectedClientBlocks, selectedActiveClientId } = useSelect(
+		( select ) => {
+			const clientID =
+				select( blockEditorStore ).getSelectedBlockClientId();
+			const selClientBlocks = select( blockEditorStore ).getBlocks();
+
+			const getBlockCommentId =
+				select( blockEditorStore ).getBlock( clientID )?.attributes
+					?.blockCommentId ?? false;
+
+			return {
+				selectedClientBlocks: selClientBlocks,
+				selectedActiveClientId: getBlockCommentId || null,
+			};
+		},
+		[]
+	);
+
+	useEffect( () => {
+		setBlocksList( selectedClientBlocks );
```

Comment:
Why the extra local state here? It's not necessary?

Reply:
The line `setBlocksList( selectedClientBlocks ); `ensures the local blocksList state stays synchronized with the selected blocks from useSelect, enabling efficient block-related operations. Without it, the component would rely directly on selectedClientBlocks, risking inconsistent data or unnecessary re-renders.

Reply:
> Without it, the component would rely directly on selectedClientBlocks, risking inconsistent data or unnecessary re-renders.

Can you elaborate more about this or how to reproduce the issue?

As far as I can see, there's no need to keep a copy of `selectedClientBlocks` in a state.

Reply:
@ellatrix and @Mamaduka , Yes I have removed the extra local state and updated the code.